src/commandsAndMenu.tsx

```diff
@@ -479,24 +480,53 @@ export function addCommands(
               tooltip: trans.__('Refresh diff widget'),
               className: 'jp-git-diff-refresh'
             });
-            refreshButton.hide();
-            diffWidget.toolbar.addItem('refresh', refreshButton);
 
-            const refresh = () => {
-              refreshButton.show();
-            };
-
-            model.changed.connect(refresh);
-            widget.disposed.connect(() => {
-              model.changed.disconnect(refresh);
+            const resolveButton = new ToolbarButton({
+              label: trans.__('Mark as resolved'),
+              onClick: async () => {
+                try {
+                  const resolvedFile: string = await widget.getResolvedFile();
+                  await serviceManager.contents.save(model.filename, {
+                    type: 'file',
+                    format: 'text',
+                    content: resolvedFile
+                  });
+                  await gitModel.add(model.filename);
+                  await gitModel.refresh();
+                } catch (reason) {
+                  logger.log({
+                    message: reason.message ?? reason,
+                    level: Level.ERROR
+                  });
+                } finally {
+                  diffWidget.dispose();
+                }
+              },
+              tooltip: trans.__('Mark file as resolved'),
+              className: 'jp-git-diff-resolve'
             });
 
+            // Do not allow the user to refresh during merge conflicts
+            if (model.isConflict) {
+              diffWidget.toolbar.addItem('resolve', resolveButton);
+            } else {
+              refreshButton.hide();
+              diffWidget.toolbar.addItem('refresh', refreshButton);
+
+              const refresh = () => {
+                refreshButton.show();
+              };
+
+              model.changed.connect(refresh);
+              widget.disposed.connect(() => model.changed.disconnect(refresh));
+            }
+
             // Load the diff widget
             modelIsLoading.resolve();
             content.addWidget(widget);
           } catch (reason) {
             console.error(reason);
-            const msg = `Load Diff Model Error (${reason.message || reason})`;
+            const msg = `Load Diff Model Error (${reason.message ?? reason})`;
```

Comment:
I would keep the _or_ operator as an empty error message is not useful.

```suggestion
            const msg = `Load Diff Model Error (${reason.message || reason})`;
```